File: mifosng-android/src/main/java/com/mifos/objects/collectionsheet/CollectionSheetRequestPayload.java
package com.mifos.objects.collectionsheet;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by Tarun on 25-07-2017.
 */

public class CollectionSheetRequestPayload implements Parcelable {

    private int calendarId;

    private String dateFormat = "dd MMM yyyy";

    private String locale = "en";

    private String transactionDate;

    public int getCalendarId() {
        return calendarId;
    }

    public void setCalendarId(int calendarId) {
        this.calendarId = calendarId;
    }

    public String getDateFormat() {
        return dateFormat;
    }

    public void setDateFormat(String dateFormat) {
        this.dateFormat = dateFormat;
    }

    public String getLocale() {
        return locale;
    }

    public void setLocale(String locale) {
        this.locale = locale;
    }

    public String getTransactionDate() {
        return transactionDate;
    }

    public void setTransactionDate(String transactionDate) {
        this.transactionDate = transactionDate;
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.calendarId);
        dest.writeString(this.dateFormat);
        dest.writeString(this.locale);
        dest.writeString(this.transactionDate);
    }

    public CollectionSheetRequestPayload() {
    }

    protected CollectionSheetRequestPayload(Parcel in) {
        this.calendarId = in.readInt();
        this.dateFormat = in.readString();
        this.locale = in.readString();
        this.transactionDate = in.readString();
    }

    public static final Parcelable.Creator<CollectionSheetRequestPayload> CREATOR = new
            Parcelable.Creator<CollectionSheetRequestPayload>() {
        @Override
        public CollectionSheetRequestPayload createFromParcel(Parcel source) {
            return new CollectionSheetRequestPayload(source);
        }

        @Override
        public CollectionSheetRequestPayload[] newArray(int size) {
            return new CollectionSheetRequestPayload[size];
        }
    };
}

File: mifosng-android/src/main/java/com/mifos/objects/collectionsheet/CollectionSheetRequestPayload.kt
package com.mifos.objects.collectionsheet

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator

/**
 * Created by Tarun on 25-07-2017.
 */
class CollectionSheetRequestPayload() : Parcelable {
    var calendarId: Int = 0
    var dateFormat: String = "dd MMMM yyyy"
    var locale: String = "en"
    var transactionDate: String? = null

    constructor(parcel: Parcel) : this() {
        calendarId = parcel.readInt()
        dateFormat = parcel.readString() ?: ""
        locale = parcel.readString() ?: ""
        transactionDate = parcel.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(calendarId)
        dest.writeString(dateFormat)
        dest.writeString(locale)
        dest.writeString(transactionDate)
    }

    override fun toString(): String {
        return "CollectionSheetRequestPayload(calendarId=$calendarId, dateFormat='$dateFormat', locale='$locale', transactionDate=$transactionDate)"
    }

    companion object CREATOR : Parcelable.Creator<CollectionSheetRequestPayload> {
        override fun createFromParcel(parcel: Parcel): CollectionSheetRequestPayload {
            return CollectionSheetRequestPayload(parcel)
        }

        override fun newArray(size: Int): Array<CollectionSheetRequestPayload?> {
            return arrayOfNulls(size)
        }
    }
}